### PR TITLE
Prerelease updates

### PR DESCRIPTION
**Pull request by antgamdia**
### Description of the change

This PR is just performing the usual update on our dependencies as described in our release process.

### Benefits

Up-to-date deps

### Possible drawbacks

Perhaps an incompatibility? Let's our CI bespeak.

### Applicable issues

- related #5082

### Additional information

N/A

### CODE REVIEW

File: dashboard/Dockerfile [hunk 1]
```diff
@@ -12,5 +12,5 @@ COPY . /app
 RUN yarn run prettier-check && yarn run ts-compile-check
 RUN yarn run build
 
-FROM bitnami/nginx:1.22.0-debian-11-r3
+FROM bitnami/nginx:1.23.1-debian-11-r1
```
Comment (antgamdia) [inline dashboard/Dockerfile:15]: I have confirmed that the update to the branch 1.23.x is also going to be carried out by the Bitnami team near soon.

**Comment by antgamdia** [comment]
I haven't pushed any CI image as I'd rather wait for https://github.com/vmware-tanzu/kubeapps/pull/5083 to be merged.

**Comment by antgamdia** [comment]
I've been looking into upgrading also the Helm version to 3.9.X (as required by #4216, eventually). In this version, they have also bumped the k8s versions up. As long as we also update our k8s dependencies we should be fine... well, we're not:

Currently, our code is not compatible with k8s >=1.24 (including upgrading the submodule) due to the following deprecations:

- [ ] Implement a `ShutDownWithDrain` function in the `Type` type (in [this file](https://github.com/vmware-tanzu/kubeapps/blob/main/cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/cache/rate_limiting_queue.go)). This is easy, and even we might replace the `reset` function we implemented (not sure), cc @gfichtenholt, [no action required now, simply just for you to know]
- [ ] Use an updated version of the `kapp` dependency. It relies on a `coreClient.Discovery().ServerResources()` that has been finally deprecated in favor of `ServerGroupsAndResources()` in k8s 1.24.

In short, even if we perform the change on the Flux queue, we would still be blocked by the kapp dependency; in particular, [this line](https://github.com/vmware-tanzu/carvel-kapp/blob/develop/go.mod#L23). If we manually update client-go in our code, we will get several compilation errors due to a dependency rename from `googleapis/gnostic` to `google/gnostic`.

So, in short, as of today, we cannot update to Helm 3.9.X nor k8s >= 1.24 unless kapp is using a `k8s.io/client-go` >= 0.23.X

**Comment by gfichtenholt** [comment]
I can take care of ShutDownWithDrain in the next PR

**Comment by antgamdia** [comment]
> I can take care of ShutDownWithDrain in the next PR

Unfortunately, this function is not yet available in the [workqueue interface](https://pkg.go.dev/k8s.io/client-go@v0.23.9/util/workqueue) on client-go v0.22.x, as it was introduced in v0.23. That said, I don't see any problem with implementing this additional function now, this way we will have done that change once we bump it up from 0.22.x, so thanks for taking care of it!

**Comment by gfichtenholt** [comment]
> > I can take care of ShutDownWithDrain in the next PR
> 
> Unfortunately, this function is not yet available in the [workqueue interface](https://pkg.go.dev/k8s.io/client-go@v0.23.9/util/workqueue) on client-go v0.22.x, as it was introduced in v0.23. That said, I don't see any problem with implementing this additional function now, this way we will have done that change once we bump it up from 0.22.x, so thanks for taking care of it!

No problem. Will take care of it

**Review by castelblanque — APPROVED** [review]
Great, thanks!

**Comment by antgamdia** [comment]
Kapp has just released a new version, https://github.com/vmware-tanzu/carvel-kapp/releases/tag/v0.52.0, updating the k8s deps!
